lib/temporal.js

```diff
@@ -263,10 +278,32 @@ exportable.repeat = function(n, ms, callback) {
   });
 };
 
-exportable.clear = function() {
-  isProcessing = false;
-  exportable.removeAllListeners();
-  queue = {};
+exportable.clear = function(id) {
+
+  if (typeof id === "undefined") {
+    isProcessing = false;
+    queue = {};
+  } else {
+    queue = filterQueue(queue, id);
+  }
+
+  if (Object.keys(queue).length === 0) {
```

Comment:
This could be potentially expensive, add a function: 

```js
function isEmpty(o) { 
  for (var p in o) { return false; }
  return true;
}
```

And then: 

```js
if (isEmpty(queue)) {...}
```